sentry-core/src/integrations/mod.rs

```diff
@@ -15,3 +22,37 @@ pub mod env_logger;
 
 #[cfg(feature = "with_panic")]
 pub mod panic;
+
+/// Integration abstraction.
+///
+/// An Integration in sentry has two primary purposes.
+/// It can act as an *Event Source*, which will capture new events;
+/// or as an *Event Processor*, which can modify every `Event` flowing through
+/// the pipeline.
+// NOTE: we need `Any` here so that the `TypeId` machinery works correctly.
+pub trait Integration: Sync + Send + Any {
+    /// Name of this integration
+    ///
+    /// This will be added to the SDK information sent to sentry
+    fn name(&self) -> Option<&str> {
```

Comment:
What if we make this `fn name(&self) -> &'static str { std::any::type_name<Self>() }`